xdsl/ir/core.py

```diff
@@ -1560,37 +1615,48 @@ def __getitem__(self, idx: int) -> Block: ...
     def __getitem__(self, idx: slice) -> Sequence[Block]: ...
 
     def __getitem__(self, idx: int | slice) -> Block | Sequence[Block]:
-        return self._blocks[idx]
+        if isinstance(idx, int):
+            if 0 <= idx:
+                for i, b in enumerate(self):
+                    if i == idx:
+                        return b
+                raise IndexError
+            else:
+                for i, b in enumerate(reversed(self)):
+                    if -1 == i + idx:
+                        return b
+                raise IndexError
+        else:
+            # This is possible but would require a bit of work to handle complex slices
+            raise NotImplementedError("Indexing of RegionBlocks not yet implemented")
 
     def __len__(self):
-        return len(self._blocks)
+        i = 0
+        for _ in self:
```

Comment:
why this way?

Reply:
as opposed to enumerating or `while (next := next.next) is not None`?